File: CSharp/AngryProfessor/AngryProfessor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace AngryProfessor
{

    public class Program

    {
        private static String[] _arrYesNo = {};
        private static int _index = 0;

        public static void Main(string[] args)
        {
            var numberOfTestCases = Int32.Parse(Console.ReadLine());
            _arrYesNo = new string[numberOfTestCases];


            for (int i = 0; i < numberOfTestCases; i++)
            {
                if (TestCase())
                {
                    _arrYesNo[_index] = "NO";
                }
                else
                {
                    _arrYesNo[_index] = "YES";
                }
                _index++;
            }

            for (int i = 0; i < numberOfTestCases; i++)
            {
                Console.WriteLine(_arrYesNo[i]);
            }
            Console.ReadLine();
        }

        private static bool TestCase()
        {
            string[] secondLine = Console.ReadLine().Split(Convert.ToChar(" "));
            int students = Int32.Parse(secondLine[0]);
            int cancelation = Int32.Parse(secondLine[1]);

            string[] thirdLine = Console.ReadLine().Split(Convert.ToChar(" "));
            int arrived = 0;
            for (int i = 0; i < students; i++)
            {
                if (Int32.Parse(thirdLine[i]) < 1)
                    arrived++;
            }

            int badStudents= students - arrived;
            //Console.WriteLine("badStudents: " + badStudents);

            if (arrived >= cancelation)
            {
                return true;
            }
            return false;
        }

    }
}

File: Ruby/angry_professor.rb


def testCase

  arrived = 0
  0.upto($students) do |i|
    if $arrival[i].to_i < 1 then
      arrived += 1
    end
  end

  if arrived < $cancel
    return false
  else
    return true
  end

end


numberOfTestCases = gets.strip.to_i

  0.upto(numberOfTestCases) do |i|
    while i < numberOfTestCases do
      firstInput = gets.chomp.split(' ')
      $students = firstInput[0].to_i
      $cancel = firstInput[1].to_i
      $arrival = gets.chomp.split(' ').collect { |n| n = n.to_i }
      $arrYesNo = {}

      if testCase == true then
        $arrYesNo[i] = 'YES'
      else
        $arrYesNo[i] = 'NO'
      end

       i += 1
    end

    0.upto(numberOfTestCases) do |ti|
      puts $arrYesNo[ti]
    end
  end



